app/src/test/java/org/mozilla/fenix/home/WallpapersObserverTest.kt

```diff
@@ -0,0 +1,175 @@
+        val appStore = AppStore()
+        val wallpapersUseCases: WallpapersUseCases = mockk {
+            coEvery { loadBitmap(any()) } returns null
+        }
+        val observer = spyk(WallpapersObserver(appStore, wallpapersUseCases, mockk(relaxed = true))) {
+            coEvery { showWallpaper(any()) } just Runs
+        }
+
+        // Ignore the call on the real instance and call again "observeWallpaperUpdates"
+        // on the spy to be able to verify the "showWallpaper" call in the spy.
+        observer.observeWallpaperUpdates()
+        coVerify { observer.showWallpaper(Wallpaper.Default) }
+
+        val wallpaper: Wallpaper = mockk(relaxed = true)
+        appStore.dispatch(UpdateCurrentWallpaper(wallpaper))
+        appStore.waitUntilIdle()
+        coVerify { observer.showWallpaper(wallpaper) }
+    }
+
+    @Test
+    fun `WHEN the wallpaper is updated to the current one THEN don't try showing the same wallpaper again`() = runTestOnMain {
+        val appStore = AppStore()
+        val wallpapersUseCases: WallpapersUseCases = mockk {
+            coEvery { loadBitmap(any()) } returns null
+        }
+        val observer = spyk(WallpapersObserver(appStore, wallpapersUseCases, mockk(relaxed = true))) {
+            coEvery { showWallpaper(any()) } just Runs
+        }
+        // Ignore the call on the real instance and call again "observeWallpaperUpdates"
+        // on the spy to be able to verify the "showWallpaper" call in the spy.
+        observer.observeWallpaperUpdates()
+
+        val wallpaper: Wallpaper = mockk(relaxed = true)
+        appStore.dispatch(UpdateCurrentWallpaper(wallpaper))
+        appStore.waitUntilIdle()
+        coVerify { observer.showWallpaper(wallpaper) }
+
+        appStore.dispatch(UpdateCurrentWallpaper(wallpaper))
+        appStore.waitUntilIdle()
+        coVerify(exactly = 1) { observer.showWallpaper(wallpaper) }
+    }
+
+    @Test
+    fun `GIVEN no wallpaper is provided WHEN asked to show the wallpaper THEN show the current one`() = runTestOnMain {
+        val wallpaper: Wallpaper = mockk()
+        val appStore: AppStore = mockk(relaxed = true) {
+            every { state.wallpaperState.currentWallpaper } returns wallpaper
+        }
+        val observer = spyk(WallpapersObserver(appStore, mockk(relaxed = true), mockk(relaxed = true)))
+
+        observer.showWallpaper()
+
+        coVerify { observer.showWallpaper(wallpaper) }
+    }
+
+    fun `GiVEN the current wallpaper is the default one WHEN showing it THEN hide the wallpaper view`() = runTestOnMain {
+        val wallpapersUseCases: WallpapersUseCases = mockk()
+        val wallpaperView: ImageView = mockk(relaxed = true)
+        val observer = WallpapersObserver(mockk(relaxed = true), wallpapersUseCases, wallpaperView)
+
+        observer.showWallpaper(Wallpaper.Default)
+
+        verify { wallpaperView.isVisible = false }
+        verify { wallpapersUseCases wasNot Called }
```

Comment:
these tests look great! I also didn't know there was a way to check for zero calls to something in a more semantic way.